### PR TITLE
MGMT-13374: support Nutanix in kube-api

### PR DESCRIPTION
**Pull request by danielerez**
Added 'Nutanix' to PlatformType enum which is used for platform selection in the AgentClusterInstall CR (via `spec.platformType`). Note: validation is already handled on [RegisterCluster](https://github.com/openshift/assisted-service/blob/6808af2917cad45dcb370e0ae3c220b4f0f95325/internal/bminventory/inventory.go#L420).

I.e.
```
kind: AgentClusterInstall
spec:
   platformType: Nutanix
```

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [x] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: subsystem/utils_test.go [hunk 1]
```diff
@@ -480,6 +480,13 @@ func getDefaultInventory(cidr string) *models.Inventory {
 	return hwInfo
 }
 
+func getDefaultNutanixInventory(cidr string) *models.Inventory {
+	vmwareInventory := *getDefaultInventory(cidr)
```
Comment (vrutkovs) [inline subsystem/utils_test.go:484]:
```suggestion
	nutanixInventory := *getDefaultInventory(cidr)
```

File: subsystem/utils_test.go [hunk 2]
```diff
@@ -480,6 +480,13 @@ func getDefaultInventory(cidr string) *models.Inventory {
...
+	vmwareInventory.SystemVendor = &models.SystemVendor{Manufacturer: "Nutanix", ProductName: "Nutanix", Virtual: true, SerialNumber: "3534"}
```
Comment (vrutkovs) [inline subsystem/utils_test.go:485]:
```suggestion
	vmwareInventory.SystemVendor = &models.SystemVendor{Manufacturer: "Nutanix", ProductName: "AHV", Virtual: true, SerialNumber: "3534"}
```
to be closer to real-world values

**Review by osherdp — APPROVED** [review]

**Pull request merged**